site/en/docs/extensions/migrating/remote-hosted-code/index.md

```diff
@@ -0,0 +1,87 @@
+
+This is no longer supported in Manifest V3 extensions and requires additional migration steps. There are different workarounds available for embedding third-party JavaScript into your extension:
+
+-   **Bundle third-party JS into your extension:** If you are using a popular framework like React or Bootstrap which you were previously loading from an external server, you can download the minified files, add them to your project and import them locally. In some cases, libraries, such as Firebase, will fetch additional code on a by need basis at runtime. In this case, you have to download all possible dynamic imports at build time. Here is an [example script that bundles Firebase using Rollup.js][7].
+-   **Embed third party JS in a sandboxed iframe:** some third-party libraries might rely on the use of `eval` or `new Function(...)` which is not supported in Manifest V3 extensions. These operations are still supported in sandboxed iframes. For example, if you need a third party library for rendering your popup HTML, you could run the external script inside a sandboxed iframe inside your popup. Inside the iframe you can still execute arbitrary JavaScript. For more details read the [guide on sandboxed iframes][3]. However, this approach does not work if the third party code requires access to your popup’s DOM.
+-   **Move functionality to your server:** communicating with remote servers is allowed. If functionality needs to change often, moving client-side functionality to your server can be a feasible workaround.
+-   **Look for other workarounds.** If the previous approaches don’t help with your use case you might have to either find an alternative solution (i.e. migrate to a different library) or find other ways to use the library's functionality. For example, in the case of Google Analytics, you can switch to the Google measurement protocol instead of using the official remote hosted JavaScript version as described in our [Google Analytics 4 guide][8].
+
+{% Aside 'warning' %}
+If our reviewers are unable to determine the full functionality of your extension during the review process, the Chrome Web Store team may reject your submission or remove it from the store.
+{% endAside %}
+
+## Add/modify extension functionality at runtime without a new store submission {: dynamic-functionality }
+
+One of the design goals of Manifest V3 is to reduce security risks for users by being able to review all the functionality provided by an extension. The Chrome Web Store’s review team must be able to easily discern the full functionality of an extension from its submitted code. For more details see the [Chrome Web Store policy for Manifest V3][9].
+
+As a consequence, some things that were possible in Manifest V2 will no longer be possible in Manifest V3. For example, launching a new UI feature in a popup without Chrome Web Store submission is no longer possible.
```

Comment:
Do you think there's a way we can merge this paragraph with the next one? I'm a bit worried about people reading "this isn't possible" and then closing the page before realising it kind of is.

Reply:
Good point rephrased the whole paragraph and extended the sample. PTA>